test/vm_tests/transformer/impl_generation_test.dart

```diff
@@ -256,6 +256,55 @@ main() {
               reason: 'should preserve existing inheritance');
         });
 
+        test('with static PropsMeta declaration', () {
+          final transformedLine = 'static const PropsMeta meta = \$Props(FooProps);';
+
+          setUpAndGenerate('''
+           @Factory()
+           UiFactory<FooProps> Foo;
+        
+           @Props()
+           class FooProps {
+            static const PropsMeta meta = \$metaForFooProps;
+           }
+           
+           @Component()
+           class FooComponent {
+            render() => null;
+           }
+           '''
+          );
+
+          var transformedSource = transformedFile.getTransformedText();
+          expect(transformedSource, contains(transformedLine));
+        });
+
+        test('with static StateMeta declaration', () {
```

Comment:
I think we can combine the StateMeta and PropsMeta tests into 1 test, since any instance of a component will have at least a props class with a `meta` field.